src/components/MagicMarkdown.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { getTCGLink } from 'utils/Affiliate';
+
+import withAutocard from 'components/WithAutocard';
+
+const AutocardLink = withAutocard('a');
+
+const MagicMarkdown = ({ markdown, cube }) => {
+  if (markdown === undefined) {
+    return '';
+  }
+  const markdownStr = markdown.toString();
+  const split = markdownStr.split(/({[wubrgcmWUBRGCM\d/-]+}|\[\[!?\d+\]\]|%%\d+%%|\n\n)/gm);
+  return (
+    <>
+      {split.map((section, position) => {
+        if (section.startsWith('{')) {
+          const symbol = section.substring(1, section.length - 1);
+          return (
+            <img
+              key={/* eslint-disable-line react/no-array-index-key */ `symbol-${position}`}
+              src={`/content/symbols/${symbol}.png`}
+              alt={symbol}
+              className="mana-symbol"
+            />
+          );
+        }
+        if (section.startsWith('[[!')) {
+          const cardIndex = parseInt(section.substring(3, section.length - 2), 10);
+          const card = cube.cards[cardIndex];
+          return (
+            <a
+              key={/* eslint-disable-line react/no-array-index-key */ `card.cardID-${position}`}
+              href={getTCGLink(card)}
+            >
+              <img src={card.details.image_normal} className="card-img-top" alt={card.details.name} />
```

Comment:
There's a FoilCardImage component you might want to use here that has autocard and stuff